two_factor/views/core.py

```diff
@@ -332,6 +340,12 @@ def get_device(self, **kwargs):
         kwargs['name'] = 'default'
         kwargs['user'] = self.request.user
 
+        if method == 'email':
+            if self.storage.validated_step_data.get('email') and not self.request.user.email:
+                self.request.user.email = self.storage.validated_step_data.get('email').get('email')
```

Comment:
When does this get saved?

Reply:
[Here](https://github.com/Atterratio/django-two-factor-auth/blob/1afe5f11a81f01e207b11d31702c3eda0b78ed9f/two_factor/forms.py#L79-L81), after input valid token, its allows add for user only confirmed email.

Reply:
I'm move it in done method of setup view.